### ISSUE TITLE
Connection won't stay alive

### ISSUE THREAD

**Issue by Thunderkill** [schroffl/teamspeak-query#9]
The client connection will cut after 10 minutes of idle.

https://nodejs.org/api/net.html#net_socket_setkeepalive_enable_initialdelay ?

**Comment by schroffl** [comment]
I will add this later today when I have the time. If you need it right now you can call the method yourself like this:

```javascript
const TeamspeakQuery = require('teamspeak-query');
const query = new TeamspeakQuery('127.0.0.1', 10011);

query.sock.setKeepAlive(true);
```

**Comment by schroffl** [comment]
It seems like teamspeak has it's own timeout-mechanism implemented, which is independent of the underlying connection. The server probably expects a valid command in a given timeframe and if nothing is sent, closes the socket.

Therefore I guess our best option is to automatically send a command ever so often, preventing the server from assuming inactivity. A command like `version` doesn't have a huge payload, which makes it feasible for such a use case.

**Edit:** I just looked at a Java library that uses a thread to regularly send keep-alive commands ([source](https://github.com/TheHolyWaffle/TeamSpeak-3-Java-API/blob/ed15557500a21195e9baca4b8b9288b2abfcd679/src/main/java/com/github/theholywaffle/teamspeak3/KeepAliveThread.java)) too.

**Comment by schroffl** [comment]
Published in [v2.1.0](https://github.com/schroffl/teamspeak-query/releases/tag/v2.1.0).

**Comment by schroffl** [comment]
In the documentation shipped with the [Server v3.3.0-beta](https://forum.teamspeak.com/threads/136231-PreRelease-TeamSpeak-3-Server-3-3-0-Beta) [1] I found out that the query timeout is configurable via the  `query_timeout` startup argument [2]. Since it isn't mentioned in the currently stable version v3.2.0, I assume that this only applies to the upcoming one.
Adjusting your client to a custom timeout is currently not possible with this library and needs to be addressed.

[1] Thanks @bigHapiness in #13 :)
[2] 5 minutes by default
